Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.cpp

```diff
@@ -0,0 +1,48 @@
+#include "fontcontrolswidgetsform.h"
+#include "ui_fontcontrolswidgetsform.h"
+
+FontControlsWidgetsForm::FontControlsWidgetsForm(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::FontControlsWidgetsForm)
+{
+    ui->setupUi(this);
+    connect(ui->lineEdit, &QLineEdit::textEdited, this, &FontControlsWidgetsForm::lineEditTextChanged);
+    connect(ui->fontComboBox, &QFontComboBox::currentFontChanged, this, &FontControlsWidgetsForm::currentFontChanged);
+    connect(ui->pushButton, &QPushButton::clicked, this, &FontControlsWidgetsForm::buttonClicked);
+}
+
+FontControlsWidgetsForm::~FontControlsWidgetsForm()
+{
+    delete ui;
+}
+
+const QString FontControlsWidgetsForm::text()
```

Comment:
Returning const isn't best practice, it's even considered harmful because it prevents moves, IIRC.

Reply:
Fixed.